core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -200,6 +331,31 @@ boolean exists(AlluxioURI path, ExistsPOptions options)
   void free(AlluxioURI path, FreePOptions options)
       throws FileDoesNotExistException, IOException, AlluxioException;
 
+  /**
+   * Builds a mapping of {@link FileBlockInfo} to a list of {@link WorkerNetAddress} which allows a
+   * user to determine the physical location of a file stored within Alluxio. In the case where
+   * data is stored in a UFS, but not in Alluxio this function will only include a
+   * {@link WorkerNetAddress} if the block stored in the UFS is co-located with an Alluxio worker.
+   * However if there are no co-located Alluxio workers for the block, then the behavior is
+   * controlled by the {@link PropertyKey#USER_UFS_BLOCK_LOCATION_ALL_FALLBACK_ENABLED} . If
+   * this property is set to {@code true} then every Alluxio worker will be returned.
+   * Blocks which are stored in the UFS and are *not* co-located with any Alluxio worker will return
+   * an empty list. If the file block is within Alluxio *and* the UFS then this will only return
+   * Alluxio workers which currently store the block.
+   *
+   * @param path the path to get block info for
+   * @return a map of blocks to the workers whose hosts have the blocks. The blocks may not
+   *         necessarily be stored in Alluxio
+   * @throws FileDoesNotExistException if the given path does not exist
+   */
+  Map<FileBlockInfo, List<WorkerNetAddress>> getBlockLocations(AlluxioURI path)
```

Comment:
if we just want a path, that means no scheme and authority, right? Should we just make it a string, and comment (this is just the path, not the uri with the scheme and auth) on it in the javadoc?

Reply:
Well I used the identifier `path` here because that's what all of the other methods specified in the client. It should behave the same way as any of the other methods if the scheme and/or authority is passed.